Optimization Workbook - Release Version - 1.0

arthurclares opened the pull request:
## 🛠️ Description
Added the latest version of the Optimization Workbook. This workbook now includes:
-Improve AHUB queries
-Summarized all Reservations opportunities in one single tab
-Added queries to check for Idle disks Snapshots and Old disks Snapshots (+30 days)

## 🔬 How has this been tested?
- [ ] 🫰 PS -WhatIf / az validate
- [x] 👍 Manually deployed + verified
- [x] 💪 Unit tests

flanakin commented on docs/optimization-workbook/workbook/azuredeploy.json:
Did you mean to add this? This should be covered by the Bicep code, right?

arthurclares replied:
The idea was to use this while we don't have the Bicep code ready yet. Are you working to develop this code? I can give it a try as well.

flanakin replied:
Bicep code is ready :-) Go to the src/scripts directory and run `./Deploy-Toolkit optimization -Build` (I think).

arthurclares replied:
Thank your for that, you are right, I have not looked at this Bicep deployment.  
Now I understood the concept. 
I have removed the deployment.json from this PR.

flanakin commented:
@arthurclares FYI - Remove the `Needs: Attention` label when you resolve the issues and need another review.

arthurclares commented on docs/optimization-workbook/workbook/Optimization Workbook.workbook:
Both .json and .workbook should work.

flanakin replied:
The Build script requires the file to be `workbook.json`. Run `Build-Workbook optimization` to test the build script. it generates the files in the `release` folder.

flanakin replied:
Don't forget to address this one so we can close the PR.

arthurclares commented on docs/optimization-workbook/workbook/placeholder:
Github was acting weird yesterday, we don't need this.

flanakin reviewed (changes requested):
Still need to rename the workbook so the build script works.

arthurclares commented:
> Still need to rename the workbook so the build script works.

I've renamed the file.
I also updated the workbook.json at /src/workbooks/workbook.json with the same code as /docs/optimization-workbook/ workbook.json

flanakin commented on docs/optimization-workbook/workbook/workbook.json:
Did you mean to put the workbook JSON in docs? We shouldn't duplicate it.

flanakin commented on docs/optimization-workbook/README.md:
```diff
@@ -10,4 +10,19 @@
+
+Use the following link to deploy the Optimization Workbook:
+
+   <a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2Farthurclares%2FWAFOptimizationFactory%2Fmain%2FOptmizationFactoryWorkbook%2Fazuredeploy.json" target="_blank"><img src="https://aka.ms/deploytoazurebutton"/></a>
```
I don't want to link out to an external repo given it'll be different than what we have here. Let's remove this for now until we get it deployed to the Azure Quickstart Templates repo. I can do that as soon as this PR is closed.
